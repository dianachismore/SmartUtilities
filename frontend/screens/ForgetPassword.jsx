import { View, Text, TextInput, StyleSheet, TouchableOpacity } from 'react-native'
import React, { useState } from 'react'
import { Button } from 'react-native-paper'
import { useDispatch, useSelector } from 'react-redux';
import { forgetPassword } from '../redux/action';

const ForgetPassword = ({ navigation }) => {

    const [email, setEmail] = useState("");

    const { loading } = useSelector(state => state.message)

    const dispatch = useDispatch()

    const forgetHandler = async () => {
        await dispatch(forgetPassword(email))
        navigation.navigate("resetpassword")

    }

    return (
        <View
            style={{
                flex: 1,
                backgroundColor: "#fff",
                alignItems: "center",
                justifyContent: "center",
            }}
        >
            <Text style={{ fontSize: 20, margin: 0 }}>FORGOT PASSWORD?</Text>
            <Text style={{ fontSize: 15, margin: 10, marginBottom: 20,color: '#9c9c9c', textAlign: 'center', width:250}}>Enter your email address and we'll send you a code to get back into your profile</Text>
            <View style={{ width: "70%" }}>
                <TextInput
                    style={Styles.input}
                    placeholder="Email"
                    value={email}
                    onChangeText={setEmail}
                />
            </View>

            <Button
                style={Styles.btn}
                onPress={forgetHandler}
                color="#fff"
                disabled={loading}
                loading={loading}
            >
                Send Email
            </Button>



        </View>
    )
}




const Styles = StyleSheet.create({

    input: {
        backgroundColor: "#fff",
        borderWidth: 1,
        borderColor: "#b5b5b5",
        padding: 10,
        paddingLeft: 15,
        borderRadius: 15,
        marginVertical: 15,
        fontSize: 15,
    },

    btn: {
        backgroundColor: "#900",
        padding: 5,
        width: "70%",
    },
})

export default ForgetPassword