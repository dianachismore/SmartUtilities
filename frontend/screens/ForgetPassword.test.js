import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import ForgetPassword from './ForgetPassword';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

// Create a mock store
const mockStore = configureStore([]);

describe('ForgetPassword Component', () => {
  let store;
  let component;
  let mockDispatch;
  let navigationMock;

  beforeEach(() => {
    // Define the mock state with the loading property
    const mockState = { message: { loading: false } };

    // Initialize the store with the mock state
    store = mockStore(mockState);

    // Create a mock dispatch function
    mockDispatch = jest.fn();

    // Create a mock navigation object
    navigationMock = {
      navigate: jest.fn(),
    };

    // Render the component with the mock store, dispatch function, and navigation object
    component = render(
      <Provider store={store}>
        <ForgetPassword navigation={navigationMock} dispatch={mockDispatch} />
      </Provider>
    );
  });

  it('should render correctly', () => {
    expect(component).toMatchSnapshot();
  });

  it('should update email state when typing in the Email input', () => {
    const emailInput = component.getByPlaceholderText('Email');
    fireEvent.changeText(emailInput, 'test@example.com');

    // Verify if the email state has been updated correctly
    expect(emailInput.props.value).toBe('test@example.com');
  });

//   it('should dispatch forgetPassword action and navigate to resetpassword screen when Send Email button is pressed', async () => {
//     const sendEmailButton = component.getByText('Send Email');

//     // Set the email input value
//     const emailInput = component.getByPlaceholderText('Email');
//     fireEvent.changeText(emailInput, 'test@example.com');

//     // Invoke the onPress event handler manually
//     sendEmailButton.props.onPress();

//     // Verify if the dispatch function has been called with the expected action
//     expect(mockDispatch).toHaveBeenCalledWith(
//       forgetPassword('test@example.com')
//     );

//     // Verify if the navigate function has been called with the expected screen name
//     expect(navigationMock.navigate).toHaveBeenCalledWith('resetpassword');
//   });
});
