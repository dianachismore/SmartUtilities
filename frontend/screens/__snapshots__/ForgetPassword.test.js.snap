// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgetPassword Component should render correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      {
        "fontSize": 20,
        "margin": 0,
      }
    }
  >
    FORGOT PASSWORD?
  </Text>
  <Text
    style={
      {
        "color": "#9c9c9c",
        "fontSize": 15,
        "margin": 10,
        "marginBottom": 20,
        "textAlign": "center",
        "width": 250,
      }
    }
  >
    Enter your email address and we'll send you a code to get back into your profile
  </Text>
  <View
    style={
      {
        "width": "70%",
      }
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Email"
      style={
        {
          "backgroundColor": "#fff",
          "borderColor": "#b5b5b5",
          "borderRadius": 15,
          "borderWidth": 1,
          "fontSize": 15,
          "marginVertical": 15,
          "padding": 10,
          "paddingLeft": 15,
        }
      }
      value=""
    />
  </View>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "#900",
        "borderColor": "transparent",
        "borderRadius": 4,
        "borderStyle": "solid",
        "borderWidth": 0,
        "elevation": 0,
        "minWidth": 64,
        "padding": 5,
        "shadowColor": "#000000",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
        "width": "70%",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "disabled": false,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "overflow": "hidden",
          },
          {
            "borderRadius": 4,
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <Text
          numberOfLines={1}
          selectable={false}
          style={
            [
              {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              {
                "textAlign": "left",
              },
              [
                {
                  "letterSpacing": 1,
                  "marginHorizontal": 16,
                  "marginVertical": 9,
                  "textAlign": "center",
                },
                undefined,
                {
                  "textTransform": "uppercase",
                },
                {
                  "color": "#fff",
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                {
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                undefined,
              ],
            ]
          }
        >
          Send Email
        </Text>
      </View>
    </View>
  </View>
</View>
`;
